### PR TITLE
Add GET item endpoint and tests

### PR DESCRIPTION
**Pull request by rjadhav-ch**
This adds GET scud item endpoint and tests

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemController.java [hunk 1]
```diff
@@ -68,4 +75,28 @@ public ScanUponDemandItemController(final ScanUponDemandItemMapper mapper, final
...
+			LoggingUtils.getLogger().info("scan upon demand item found", logMap);
+			return ResponseEntity.status(OK).body(retrievedScanUponDemandItemResponseDTO);
+		}
+		else {
+			final String errorMsg = "scan upon demand resource not found";
+			final List<String> errors = new ArrayList<>();
+			errors.add(errorMsg);
+			LoggingUtils.logErrorsWithStatus(logMap, errors, NOT_FOUND);
+			LoggingUtils.getLogger().error(errorMsg, logMap);
+			return ResponseEntity.status(NOT_FOUND).body(new ApiError(NOT_FOUND, errors));
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemController.java:99]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/dto/ScanUponDemandItemResponseDTO.java [hunk 2]
```diff
@@ -24,6 +27,10 @@
 
 	private String etag;
 
+	private List<ItemCosts> itemCosts;
+
+	private ScanUponDemandItemOptions itemOptions;
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/dto/ScanUponDemandItemResponseDTO.java:33]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/dto/ScanUponDemandItemResponseDTO.java [hunk 3]
```diff
@@ -98,6 +105,14 @@ public void setEtag(String etag) {
...
 	}
 
+	public List<ItemCosts> getItemCosts() { return itemCosts; }
+
+	public void setItemCosts(List<ItemCosts> itemCosts) { this.itemCosts = itemCosts; }
+
+	public ScanUponDemandItemOptions getItemOptions() { return itemOptions; }
+
+	public void setItemOptions(ScanUponDemandItemOptions itemOptions) { this.itemOptions = itemOptions; }
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/dto/ScanUponDemandItemResponseDTO.java:115]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/logging/LoggingUtils.java [hunk 4]
```diff
@@ -39,4 +41,17 @@ public static Logger getLogger() {
...
+	 * purposes
+	 * @param logMap the map of logging data
+	 * @param errors a list of errors
+	 */
+	public static void logErrorsWithStatus(final Map<String, Object> logMap,
+										   final List<String> errors,
+										   final HttpStatus status) {
+		logMap.put(ERRORS_LOG_KEY, errors);
+		logMap.put(STATUS_LOG_KEY, status);
+	}
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/logging/LoggingUtils.java:56]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItem.java [hunk 5]
```diff
@@ -82,11 +82,15 @@ public void setCustomerReference(String companyReference) {
 	public void setEtag(String etag) {
 		data.setEtag(etag);
 	}
-	
+
+	public void setItemOptions(ScanUponDemandItemOptions itemOptions) { data.setItemOptions(itemOptions); }
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItem.java:87]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItem.java [hunk 6]
```diff
@@ -82,11 +82,15 @@ public void setCustomerReference(String companyReference) {
...
 	public void setLinks(Links links) {
 		data.setLinks(links);
 	}
 
+	public void setPostageCost(String postageCost) { data.setPostageCost(postageCost); }
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItem.java:93]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItemData.java [hunk 7]
```diff
@@ -28,6 +29,10 @@
 
 	private String etag;
 
+	private List<ItemCosts> itemCosts;
+
+	private ScanUponDemandItemOptions itemOptions;
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItemData.java:35]: Tabs need to be replaces with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItemData.java [hunk 8]
```diff
@@ -102,6 +107,14 @@ public void setEtag(String etag) {
...
 	}
 
+	public List<ItemCosts> getItemCosts() { return itemCosts; }
+
+	public void setItemCosts(List<ItemCosts> itemCosts) { this.itemCosts = itemCosts; }
+
+	public ScanUponDemandItemOptions getItemOptions() { return itemOptions; }
+
+	public void setItemOptions(ScanUponDemandItemOptions itemOptions) { this.itemOptions = itemOptions; }
+
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/model/ScanUponDemandItemData.java:117]: Tabs need to be replaced with spaces

File: src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/service/ScanUponDemandItemService.java [hunk 9]
```diff
@@ -52,4 +53,12 @@ void setCreationDateTimes(final ScanUponDemandItem item) {
...
 	}
 
+	/**
+	 * Returns scan upon demand item
+	 * @param id id of the requested item
+	 * @return
+	 */
+	public Optional<ScanUponDemandItem> getScanUponDemandItemById(String id) {
+		return repository.findById(id);
+	}
```
Comment (hwilliams1) [inline src/main/java/uk/gov/companieshouse/scanupondemand/orders/api/service/ScanUponDemandItemService.java:63]: Tabs need to be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/HealthcheckControllerIntegrationTest.java [hunk 10]
```diff
@@ -20,6 +20,6 @@
 	@Test
 	@DisplayName("Successfully returns health status")
 	public void returnHealthStatusSuccessfully() throws Exception {
-		mockMvc.perform(get("/orderable/scans-upon-demand/healthcheck")).andExpect(status().isOk());
+		mockMvc.perform(get("/orderable/scan-upon-demand/healthcheck")).andExpect(status().isOk());
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/HealthcheckControllerIntegrationTest.java:23]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java [hunk 11]
```diff
@@ -10,61 +10,70 @@
...
-
 	private static final Links LINKS;
+	private static final String TOKEN_ETAG = "9d39ea69b64c80ca42ed72328b48c303c4445e28";
+	private static final String POSTAGE_COST = "0";
+
+	private static final String FILING_HISTORY_ID = "1";
+	private static final String FILING_HISTORY_DATE = "2010-02-12";
+	private static final String FILING_HISTORY_DESCRIPTION = "change-person-director-company-with-change-date";
+	private static final Map<String, Object> FILING_HISTORY_DESCRIPTION_VALUES;
+	public static final String FILING_HISTORY_TYPE_CH01 = "CH01";
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java:58]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java [hunk 12]
```diff
@@ -10,61 +10,70 @@
...
 
 	static {
 		LINKS = new Links();
 		LINKS.setSelf(SCAN_UPON_DEMAND_URL + "/" + SCAN_UPON_DEMAND_ID);
+		FILING_HISTORY_DESCRIPTION_VALUES = new HashMap<>();
+		FILING_HISTORY_DESCRIPTION_VALUES.put("change_date", "2010-02-12");
+		FILING_HISTORY_DESCRIPTION_VALUES.put("officer_name", "Thomas David Wheare");
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java:65]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java [hunk 13]
```diff
@@ -95,6 +103,66 @@ void createScanUponDemandItemSuccessfullyCreatesScanUponDemandItem() throws Exce
...
+	void getScanUponDemandItemReturnsNotFound() throws Exception {
+		// When and then
+		mockMvc.perform(get(SCAN_UPON_DEMAND_URL + "/" + SCAN_UPON_DEMAND_ID)
+				.header(REQUEST_ID_HEADER_NAME, REQUEST_ID_VALUE)
+				.header(REQUEST_ID_HEADER_NAME, REQUEST_ID_VALUE)
+				.contentType(MediaType.APPLICATION_JSON))
+				.andExpect(status().isNotFound())
+				.andDo(MockMvcResultHandlers.print());
+	}
+
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerIntegrationTest.java:165]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java [hunk 14]
```diff
@@ -39,14 +39,29 @@
 	private ScanUponDemandItemData data;
 
 	@Mock
-	private ScanUponDemandItemRequestDTO dto;
+	private ScanUponDemandItemResponseDTO dto;
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java:42]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java [hunk 15]
```diff
@@ -39,14 +39,29 @@
...
-	void test1() {
+	@DisplayName("GET scan upon demand resource returns item")
+	void getScanUponDemandItemPresent() {
+		when(scanUponDemandItemService.getScanUponDemandItemById(ID)).thenReturn(Optional.of(item));
+		when(item.getData()).thenReturn(data);
+		when(mapper.scanUponDemandItemToScanUponDemandItemResponseDTO(data)).thenReturn(dto);
+		ResponseEntity<Object> response = controllerUnderTest.getScanUponDemandItem(ID, REQUEST_ID_VALUE);
+
+		assertThat(response.getStatusCode(), is(HttpStatus.OK));
+		assertThat(response.getBody(), is(dto));
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java:56]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java [hunk 16]
```diff
@@ -39,14 +39,29 @@
...
 	}
 
+	@Test
+	@DisplayName("Get scan upon demand resource returns HTTP NOT FOUND")
+	void getCertifiedCopyItemNotFound() {
+		when(scanUponDemandItemService.getScanUponDemandItemById(ID)).thenReturn(Optional.empty());
+		ResponseEntity<Object> response = controllerUnderTest.getScanUponDemandItem(ID, REQUEST_ID_VALUE);
+
+		assertThat(response.getStatusCode(), is(HttpStatus.NOT_FOUND));
+	}
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/controller/ScanUponDemandItemControllerTest.java:66]: Tabs should be replaced with spaces

File: src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/util/TestConstants.java [hunk 17]
```diff
@@ -3,5 +3,5 @@
 public class TestConstants {
 	public static final String REQUEST_ID_HEADER_NAME = "X-Request-ID";
 	public static final String REQUEST_ID_VALUE = "f058ebd6-02f7-4d3f-942e-904344e8cde5";
-	public static final String SCAN_UPON_DEMAND_URL = "/orderable/scans-upon-demand";
+	public static final String SCAN_UPON_DEMAND_URL = "/orderable/scan-upon-demand";
```
Comment (hwilliams1) [inline src/test/java/uk/gov/companieshouse/scanupondemand/orders/api/util/TestConstants.java:6]: Tabs should be replaced with spaces

**Review by hwilliams1 — CHANGES_REQUESTED** [review]
Can we change the tabs for spaces please? This should be done for all of the code containing tabs that are included with this PR. The re-formatting of the remaining code can be done as another Mira task

**Comment by rjadhav-ch** [comment]
> Can we change the tabs for spaces please? This should be done for all of the code containing tabs that are included with this PR. The re-formatting of the remaining code can be done as another Mira task

I've replaced the tabs with spaces.

**Review by hwilliams1 — APPROVED** [review]
Looks OK

**Review by VaughanJackson123 — APPROVED** [review]

**Pull request merged**